### PR TITLE
solution

### PR DESCRIPTION
**Pull request by serentinos**
Solution

### CODE REVIEW

File: src/createServer.js [hunk 1]
```diff
@@ -1,3 +1,50 @@
...
+        res.end();
+
+        return;
+      }
+
+      const textToTransform = normalizedUrl.pathname.slice(1);
+      const toCaseTransform = normalizedUrl.searchParams.get('toCase');
+      const isValidData = validateRequest(textToTransform, toCaseTransform);
+
+      if (!isValidData.isValid) {
```
Comment (33Orange) [inline src/createServer.js:22]:
```suggestion
      const errors = validateRequest(textToTransform, toCaseTransform);

      if (errors.length) {
```

File: src/createServer.js [hunk 2]
```diff
@@ -1,3 +1,50 @@
...
+        const error = { errors: isValidData.errorArray };
+
+        res.writeHead(400, 'Bad reguest', {
```
Comment (33Orange) [inline src/createServer.js:25]:
```suggestion
        res.writeHead(400, 'Bad request', {
```

File: src/createServer.js [hunk 3]
```diff
@@ -1,3 +1,50 @@
...
+          'content-type': 'application/json',
+        });
+        res.write(JSON.stringify(error));
+        res.end();
```
Comment (33Orange) [inline src/createServer.js:29]: You can provide a response in the `end()` method

```suggestion
        res.end(JSON.stringify(error));
```

File: src/createServer.js [hunk 4]
```diff
@@ -1,3 +1,50 @@
...
+        return;
+      }
+
+      const resultRespond = generateRespond(textToTransform, toCaseTransform);
+
+      res.writeHead(200, 'OK', {
+        'content-type': 'application/json',
+      });
+      res.write(JSON.stringify(resultRespond));
+      res.end();
```
Comment (33Orange) [inline src/createServer.js:40]: Same

**Review by 33Orange — CHANGES_REQUESTED** [review]
Almost done

File: src/createServer.js [hunk 5]
```diff
@@ -1,3 +1,50 @@
...
+
+  const server = http.createServer((req, res) => {
+    try {
+      const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+
+      if (normalizedUrl.pathname === faviconURL) {
+        res.end();
+
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (serentinos) [reply src/createServer.js:16]: the browser try to do reguest on /favicon.ico. I've tried to ignore server response on that request

**Comment by serentinos** [comment]
@33Orange Thank you for the suggestions! The code is much cleaner now!

**Review by 33Orange — APPROVED** [review]
Well done

**Pull request closed without merging**